### PR TITLE
[query] Add or_error method to SwitchBuilder

### PR DESCRIPTION
**Pull request by nawatts**
CaseBuilder has an `or_error` method to throw an error if no `when` conditions are true. Currently, SwitchBuilder does not have an equivalent method: it only supports returning a default value or missing. The option to throw an error on an unhandled value can be useful for making sure that all possible values for an enum expression have been accounted for in switch cases.

### CODE REVIEW

File: hail/python/hail/expr/builders.py [hunk 1]
```diff
@@ -83,7 +83,7 @@ def f(base):
     @typecheck_method(value=expr_any, then=expr_any)
     def when(self, value, then) -> 'SwitchBuilder':
         """Add a value test. If the `base` expression is equal to `value`, then
-         returns `then`.
+        returns `then`.
```
Comment (nawatts) [inline hail/python/hail/expr/builders.py:86]: Unrelated formatting fix for [SwitchBuilder.when docs](https://hail.is/docs/0.2/functions/hail.expr.builders.SwitchBuilder.html#hail.expr.builders.SwitchBuilder.when).

**Review by johnc1231 — APPROVED** [review]
Thanks, good addition. The one thing I'd say is just that when we make a user facing change, we try to make one of the commits look like:

CHANGELOG: Added `or_error` method to `SwitchBuilder`

This ends up being helpful when I have to go through and generate the change log for a new version release.

**Comment by nawatts** [comment]
> The one thing I'd say is just that when we make a user facing change, we try to make one of the commits look like:
>
> CHANGELOG: Added or_error method to SwitchBuilder
>
> This ends up being helpful when I have to go through and generate the change log for a new version release.

Good to know. It would be nice to document conventions like this in the [docs for software developers](https://hail.is/docs/0.2/getting_started_developing.html#contributing) and/or a [contributing file](https://docs.github.com/en/free-pro-team@latest/github/building-a-strong-community/setting-guidelines-for-repository-contributors).

**Comment by johnc1231** [comment]
This is a good point, we should look into doing that. Thanks for the suggestion.

**Comment by nawatts** [comment]
Added the "CHANGELOG" note to contribution guidelines in #9752.
